docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+[[analysis-synonyms]]
+== Synonyms
+
+Synonyms are words or phrases that have the same or similar meaning.
+They are an important aspect of search, as they can improve the search experience and the relevancy of search results.
+
+Synonyms allow you to:
+
+* Improve search relevance: Synonyms enable users to find relevant documents even if they use different terms to express the same concept.
+* Increase search scope: Using synonyms ensures a wider range of possible terms. Search results are not limited to the specific terms included in your documents, and variations of relevant terms are included in search results.
+* Domain-specific vocabulary: Certain domains use specific terms that users are not familiar with. Using synonyms allow search to become more user-friendly, as users can search terms they are familiar with and search results will not be limited to the specific domain vocabulary.
+* Handling common misspellings and typos: Synonyms can be used to make sure that frequent or specific misspellings are correctly handled.
+
+Synonyms are grouped together using *synonyms sets*.
```

Comment:
Is this supposed to be a general document about "synonyms"?  Why we are talking here specifically about "synonyms set"?